source/ncengine/graphics2/diligent/pass/PassUtilities.cpp

```diff
@@ -132,6 +132,27 @@ void BindRenderTarget(Diligent::IDeviceContext& context,
     context.SetRenderTargets(colorRenderTargetIndex == NoTarget ? 0 : 1, &pRTV, pDSV, Diligent::RESOURCE_STATE_TRANSITION_MODE::RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
 }
 
+auto CreateShaderFromSourceIfInitialized(ShaderFactory& shaderFactory, Diligent::SHADER_TYPE shaderType, const nc::graphics::ShaderPaths& shaderPaths) -> Diligent::RefCntAutoPtr<Diligent::IShader>
+{
+    Diligent::RefCntAutoPtr<Diligent::IShader> shader;
+    const auto& shaderPath = shaderType == Diligent::SHADER_TYPE_VERTEX ? shaderPaths.vertexShaderPath : shaderPaths.pixelShaderPath;
+
+    if (shaderPath == "Uninitialized")
```

Comment:
Either way, "Uninitialized" is a surprising 'empty' state to me. Can we default initialize to an empty string instead?

Reply:
Yes - you're right that's much more sane